lib/Macros/Sources/SwiftMacros/TaskLocalMacro.swift

```diff
@@ -0,0 +1,204 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2022-2023 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import SwiftSyntax
+import SwiftSyntaxMacros
+import SwiftDiagnostics
+
+/// Macro implementing the TaskLocal functionality.
+///
+/// It introduces a peer `static let $name: TaskLocal<Type>` as well as a getter
+/// that assesses accesses the task local storage.
+public enum TaskLocalMacro {}
+
+extension TaskLocalMacro: PeerMacro {
+  public static func expansion(
+    of node: AttributeSyntax,
+    providingPeersOf declaration: some DeclSyntaxProtocol,
+    in context: some MacroExpansionContext
+  ) throws -> [DeclSyntax] {
+    guard let varDecl = try requireVar(declaration, diagnose: false) else {
+      return []
+    }
+    guard try requireStaticContext(varDecl, in: context, diagnose: false) else {
+      return []
+    }
+
+    guard let firstBinding = varDecl.bindings.first else {
+      throw DiagnosticsError(
+        syntax: declaration,
+        message: "'@TaskLocal' property must have declared binding", id: .incompatibleDecl)
+    }
+
+    guard let name = firstBinding.pattern.as(IdentifierPatternSyntax.self)?.identifier else {
+      throw DiagnosticsError(
+        syntax: declaration,
+        message: "'@TaskLocal' property must have name", id: .incompatibleDecl)
+    }
+
+    let type = firstBinding.typeAnnotation?.type
+    let explicitType: String
+    if let type {
+      explicitType = ": TaskLocal<\(type.trimmed)>"
+    } else {
+      explicitType = ""
+    }
+
+    let initialValue: Any
+    if let initializerValue = firstBinding.initializer?.value {
+      initialValue = initializerValue
+    } else if let type, type.isOptional {
+      initialValue = "nil"
+    } else {
+      throw DiagnosticsError(
+        syntax: declaration,
+        message: "'@TaskLocal' property must have default value, or be optional", id: .mustBeVar)
+    }
+
+    // If the property is global, do not prefix the synthesised decl with 'static'
+    let isGlobal = context.lexicalContext.isEmpty
+    let staticKeyword: String
+    if isGlobal {
+      staticKeyword = ""
+    } else {
+      staticKeyword = "static "
+    }
```

Comment:
Similar here, you could just `TokenSyntax` instead of strings

```suggestion
    let staticKeyword: TokenSyntax?
    if isGlobal {
      staticKeyword = nil
    } else {
      staticKeyword = TokenSyntax.keyword(.static, trailingTrivia: .space)
    }
```